saleor/dashboard/order/forms.py

```diff
@@ -60,13 +60,14 @@ def clean(self):
     def payment_error(self, message):
         self.add_error(
             None, pgettext_lazy(
-                'Payment form error',
-                'Payment gateway error: %s') % message)
+                'Payment form error', 'Payment gateway error: %s') % message)
 
     def try_payment_action(self, action):
-        amount = self.cleaned_data['amount']
+        # Unwrap Money() object to Decimal before passing it
+        # to payments utility library for processing
+        amount = self.cleaned_data['amount'].amount
```

Comment:
I think this would be less confusing had it stayed in its previous form:

```python
money = self.cleaned_data['amount']
try:
    action(money.amount)
...
```